test/unit/data/events/add-pending-events.test.js

```diff
@@ -0,0 +1,65 @@
+const { PAYMENT_ENRICHED_NAME, PAYMENT_PROCESSED_NAME, PAYMENT_SUBMITTED_NAME, PAYMENT_ACKNOWLEDGED_NAME } = require('../../../../app/constants/names')
+
+const enriched = require('../../../mocks/events/enriched')
+const processed = require('../../../mocks/events/processed')
+const submitted = require('../../../mocks/events/submitted')
+const acknowledged = require('../../../mocks/events/acknowledged')
+
+const { addPendingEvents } = require('../../../../app/data/events/add-pending-events')
+
+let groupedEvent
+
+describe('add pending events', () => {
+  beforeEach(() => {
+    groupedEvent = JSON.parse(JSON.stringify(require('../../../mocks/events/grouped-event')))
+  })
+
+  test('should add all default events if no events', () => {
+    const result = addPendingEvents([groupedEvent])
+    expect(result[0].events).toHaveLength(4)
+  })
+
+  test('should add enrichment event if no enrichment event', () => {
```

Comment:
Could we change each of the 'should add _ event if no _ event' to 'should add _ event if no _ event exists' just to make it a bit clearer?